src/js/components/FocusedContainer.js

```diff
@@ -12,14 +12,114 @@ export const FocusedContainer = ({
 }) => {
   const [bodyOverflowStyle, setBodyOverflowStyle] = useState('');
   const ref = useRef(null);
+  const preNodeRef = useRef(null);
+  const postNodeRef = useRef(null);
 
-  const roots = useContext(RootsContext);
-  const [nextRoots, setNextRoots] = useState(roots);
+  const { roots, setRoots } = useContext(RootsContext);
+  const [active, setActive] = useState(true);
+
+  // only the most recent container in roots should be active and listening
+  // for focus events. when active is false, event listeners will be removed
   useEffect(() => {
-    // make sure value of null is not added to array
-    if (ref.current) setNextRoots([...roots, ref.current]);
+    if (ref.current && roots[roots.length - 1] === ref.current) {
+      setActive(true);
+    } else setActive(false);
   }, [roots]);
 
+  useEffect(() => {
+    const container = ref.current;
+    if (container) {
+      // avoid infinite loop by only adding it if it's not already there
+      const nextRoots = !roots.includes(container)
+        ? [...roots, container]
+        : roots;
+      setRoots(nextRoots);
```

Comment:
// try this approach to avoid the infinite loop

setRoots(prevValue => {

});